py3status/modules/wwan.py

```diff
@@ -342,38 +343,80 @@ def _get_bearer(self, modem_proxy):
             pass
         return bearer
 
-    def _get_messages(self, modem_proxy):
+    def _organize_messages(self, messages):
         messages_data = {}
-        key = 0
-        try:
-            messages = modem_proxy.Messages
-            # loop on sms list for organize them
-            sms_data = {}
+        if messages:
+            sms_data = []
             for msg in messages:
                 id = int(msg.replace(STRING_MODEMMANAGER_SMS_PATH, ''))
                 sms_proxy = self.bus.get(STRING_MODEMMANAGER_DBUS, msg)
-                contact = sms_proxy.SMSC
+                contact = sms_proxy.Number
                 text = sms_proxy.Text
-                key += 1
-                sms_data[key] = {'id': id, 'contact': contact, 'text': text}
+                text_short = self._truncate_message(text)
                 messages_data['last_contact'] = '{}'.format(contact)
                 messages_data['last_text'] = '{}'.format(text)
                 messages_data['last_text_short'] = '{}'.format(
-                    self._trunc_message(text))
-                messages_data['last_message'] = '{}: {}'.format(contact, text)
-                messages_data['last_message_short'] = '{}: {}'.format(
-                    contact, self._trunc_message(text))
+                    self._truncate_message(text))
+                sms_data.insert(
+                    0, {
+                        'id': id,
```

Comment:
(Q) Is `id` more of message unique id or message index?

Reply:
It is not an unique id, not a id, seems a temporary id of SMS content.

On each modem down/up/laptop restart etc 'modem init' maybe is the good term, we have an id, and i need this id to query with dbus the message content:
```
      └─/org/freedesktop/ModemManager1/SMS
        ├─/org/freedesktop/ModemManager1/SMS/546
        ├─/org/freedesktop/ModemManager1/SMS/547
        ├─/org/freedesktop/ModemManager1/SMS/548
        ├─/org/freedesktop/ModemManager1/SMS/549
        ├─/org/freedesktop/ModemManager1/SMS/550
        ├─/org/freedesktop/ModemManager1/SMS/551
        ├─/org/freedesktop/ModemManager1/SMS/552
        ├─/org/freedesktop/ModemManager1/SMS/553
        ├─/org/freedesktop/ModemManager1/SMS/554
        ├─/org/freedesktop/ModemManager1/SMS/555
        ├─/org/freedesktop/ModemManager1/SMS/556
...
```
and so I have with array (oups, List in python), i would like to query the last with [-1]

Reply:
>i would like to query the last with [-1]

Assuming users will never care about last `_3_` messages, we could `enumerate` the list and introduce new placeholder `index`, reverse sort the list, then users can adjust number of last messages they want to see... eg
* `format_message = '[\?if=index<4 {message}]'` to display last `_3_` messages (and)
* `format_message = '[\?if=index=1 {message}]'` to display last `_1_` message.

If users really want to view (first) oldest messages, then we could expose a new config `sort` to sort the list which I think is not going to be used much (possibly no usage), so probably better if we don't do that until it's needed first. More flexible (imho).